packages/runner/src/test-state.ts

```diff
@@ -9,3 +9,15 @@ export function setCurrentTest<T extends Test>(test: T | undefined): void {
 export function getCurrentTest<T extends Test | undefined>(): T {
   return _test as T
 }
+
+const tests: Array<Test> = []
+export function addRunningTest(test: Test): () => void {
+  const index = tests.push(test)
+  return () => {
+    tests.splice(index)
+  }
```

Comment:
Is it possible that this is not called serially, e.g. in `test.concurrent`? The stored `index` could become stale.

```suggestion
  return () => {
    tests.splice(tests.indexOf(test))
  }
```

Reply:
True, this code won't work with concurrent tests